[REFACTOR] : 변경된 Entity와 유저 로그인 Confilct 해소 및 merge

twotwobread opened the pull request:
<!-- 이슈와 PR 이름을 맞춰주세요!! -->
<!-- 여러 라인에 걸쳐 문장이 필요한 경우, 각 라인마다 "-" 를 이용해주세요 -->

## PR Description 🗒️
<!-- 
PR에 대한 간략한 설명을 적어주세요
ex)
- 로그인 기능 추가.
-->
close #71 
close #11 #27   
close #12 #13 #76 #88 #95 #107 
- 로그인/회원가입 소셜 로그인을 위한 기능 개발
- 유저 조회 관련 기능 개발 및 테스트 코드 작성.
- log 관련 설정
## 추가 사항 및 설명 ➕
<!-- 
[FEAT] / [ADD]
추가 사항과 이에 대한 설명을 적어주세요
ex)
- 예매 기능 추가
    - 
-->
- 소셜(구글) 로그인/회원가입 기능 개발 - [twotwobread](https://github.com/twotwobread), [park-se-jun](https://github.com/park-se-jun) 페어프로그래밍
  - SpringSecufityConfig 설정
    - access deny handler 설정. 
    - 정적 파일에 대해 인가 수행하지 않게 구성하고 User 권한을 가진 사람만 /api/v1/** url로 접근할 수 있게 구성.
    - oauth2Login을 이용하게 구성하고 authorization endpoint, authorizationRequestRepository, redirect endpoint, user info endpoint 설정.
    - success handler 설정.
  - 권한 확인 후 엑세스 거부되었을 때 실행될 OAuthAccessDeniedHandler  추가.
    - 상태코드 403과 함께 간단한 text 출력.
  - CookieOAuth2AuthorizationRequestRepository와 CookieUtil 추가.
    - request 정보로부터 Cookie를 가져오는 기능.
    - 쿠키를 저장하는 기능.
    - 쿠키를 삭제하는 기능.
  - CustomIOAuth2UserService와 CustomUserPrincipal 추가.
    - Resource 서버 (구글)로부터 User의 허용된 정보들을 가져오는 부분.
    - 해당 소셜 상의 id, 유저 이름, 유저 역할, 이메일 등의 정보를 가져옴.
    - 해당 정보로 OAuth2User의 구현체인 CustomUserPrincipal 내부 필드값으로 할당하여 반환.
  - success handler인 OAuth2AuthenticationSuccessHandler 추가
    -  실질적으로 우리의 서비스에 유저를 회원가입하는 부분.
      - UserService의 join을 메소드를 이용해서 이미 유저가 존재하면 유저의 변경가능 정보인 user nickname을 업데이트하고 해당 유저 정보를 반환.
      - 유저가 존재하지 않으면 db에 저장하고 해당 유저 정보를 반환
    - 반환된 유저 정보를 이용하여 access token과 refresh token 생성 및 기존 쿠키를 삭제하고 새로운 refresh token 정보를 쿠키에 저장.
    - 인증 성공 반환 정보로 access token을 반환.
  - TokenProvider 추가.
    - access, refresh 토큰을 생성하기 위한 로직 존재.
  - TokenAuthenticationFilter 추가.
    - access toekn과 함께 요청이 들어올 경우 validate 수행하고 인증이 된것이 맞으면 Authentication을 생성하고 SpringSecutiryContext에 저장.
    - 인증되지 않은 경우 그냥 다음 filter 수행.
  - OAuthController
    - 만료된 토큰 refresh를 위해서 /refresh url 컨트롤러 추가. -> 갱신된 토큰 반환.
- 유저 상세 정보 조회 기능 - [twotwobread](https://github.com/twotwobread)
  - isDeleted 상태를 고려하여 repository, service, controller 구현.
  - 성공/실패 관련 test 코드 구현.
- 유저 페이징 조회 기능 - [twotwobread](https://github.com/twotwobread)
  - 현재 페이지 번호가 가능한 마지막 페이지를 넘는 경우를 고려하여 repository, service, controller 구현.
  - 성공/실패 관련 test 코드 구현.
- log 관련 설정. - [twotwobread](https://github.com/twotwobread)
  - logback 로그 메시지 패턴 설정.
  - aws clock watch에 해당 로그가 쌓일 수 있게 설정. 

## To Reviewers 🙏
<!-- 
리뷰어들이 중점적으로 봐줬으면 하는 부분이나 
리뷰어들에게 추가적으로 하고 싶은 말을 적어주세요!! 
-->
- squash and merge 기능 때문에 커밋이 안보일꺼 같애서 최대한 자세히 적었습니다!!

twotwobread committed:
[FEAT] : 스프링 설정 파일과 ignore을 추가.

Close feature/이수영_#27

twotwobread committed:
[ADD] : OAuth2 관련 의존성 추가.

Close feature/이수영_#27

twotwobread committed:
[ADD] : User 관련 필드 수정 및 UserRole enum 클래스 필드 수정.

- User 필드에 이메일, 패스워드를 추가하고 social enum 클래스를 SocialType으로 리네이밍.
- UserRole 내부에 key와 title 필드를 추가.

Close feature/이수영_#27

twotwobread committed:
[FEAT] : UserRepository 추가.

- User CRUD를 위해 JpaRepository를 상속한 Repository 인터페이스 추가.
- unique한 email을 이용하여 User find 쿼리 메소드 추가.

Close feature/이수영_#27

twotwobread committed:
[FEAT] : 스프링 시큐리티 관련 설정 파일 추가.

Close feature/이수영_#27

twotwobread committed:
[FEAT] : Authorization Server에서 받아온 attributes들을 이용하여 User 엔티티를 만들기 위한 클래스 추가.

Close feature/이수영_#27

twotwobread committed:
[FEAT] : Authorization Server로 부터 가져온 attributes를 이용하여 회원가입, 로그인 기능 추가.

- CustomOauth2UserService에서 가져온 attributes를 이용하여 클라이언트 서비스에 User가 존재하는지 확인 후 없다면 회원가입, 존재하면 로그인 기능 추가.
- CustomUserPrincipal을 통해 attributes 정보와 현재 서비스에 생성 혹은 존재하는 User 정보를 담을 클래스 추가.

Close feature/이수영_#27

park-se-jun committed:
feat : customeOauth2userService에서 처리하던 User save 로직을 OAuth2AuthenticationSuccessHandler로 이동
- successHandler 추가 및 config 설정
- UserService에 user를 save하는 로직 추가
- userRepository에 socialId와 social type으로 user를 꺼내오는 메소드 추가
- CustomUserPrincipal에 getName을 socialId를 반환하도록 변경(디폴트가 그거인듯)

park-se-jun committed:
[FEAT] : accessToken, refreshToken 생성, db에 저장

- Token을 생성하는 TokenProvider 작성
- refreshToken 및 accessToken 생성
- refreshToken db에 저장

twotwobread committed:
[FEAT] : tokenProvider를 이용하여 유저 존재 여부 및 refresh 토큰과의 id가 같은지 검증하는 로직 추가.

Close feature/이수영_#27

twotwobread committed:
[FEAT] : token 관련 간단한 예외 핸들러 추가.
- refresh token을 이용하여 유저 조회 시 유저를 찾지 못할 경우에 대한 예외 핸들링 추가.

Close feature/이수영_#27

twotwobread committed:
[FEAT] : 쿠키에 refresh 토큰을 저장하기 위한 repository 추가.

- refresh 토큰을 쿠키에 설정하기 위한 repository 추가.
Close feature/이수영_#27

park-se-jun committed:
[FEAT] : CookieUtils 추가

- 이름으로 지우는 deleteCookieByName 작성
- 이름으로 쿠키를 찾는 getCookieByName 작성
- response에 cookie를 추가하는 addCookie 작성

park-se-jun committed:
[FEAT] : Successhandler에 refreshToken와 AessesToken을 다루는 로직 추가

- SuccessHander에 refreshToken을 cookie로 추가하는 로직 추가
- SuccessHandler에 accessToken을 responseBody로 추가하는 로직 추가
- TokenProvider와 SuccessHander의 상수들을 JwtConstants로 이동

park-se-jun committed:
Merge remote-tracking branch 'origin/feature/이수영_#27' into feature/이수영_#27

park-se-jun committed:
[FIX] : 순환참조 해결

- userService가 SpringSecurityConfig의 TokenProvider를 참조하는데, SpringSecurityConfig가 userService를 참조해서 발생하는 문제 해결

park-se-jun committed:
[CHORE] : CookieOAuth2AuthorizationRequestRepository에 CookieUtils 사용 및 Constant 분리
- CookieUtils를 사용하도록 코드 변경
- Constant를 분리

park-se-jun committed:
[FIX] : merge 과정에서 반영되지 않은 userService.assignRefreshToken 변경 반영
- userId와 refreshToken을 받는 시그니처에서 refreshToken만 받는 시그니쳐로 변경된 사항 반영

twotwobread committed:
[ADD] : UserIdANdRoleDTO 추가에 따른 로직 변경.

- UserIdAndRoleDto를 추가함 ( userId와 userRole에 대한 정보를 담고 있음)
- UserService에서도 이를 반환하게 수정함.
CLose feature/이수영_#27

twotwobread committed:
[FEAT] : TokenProvider 기능 추가.

- validate를 이용하여 accessToken의 유효성 검증하는 메소드 추가.
- accessToken을 이용하여 Authentication을 가져오는 메소드 추가.
- token 생성 시에 UserRole 정보 추가.
Close feature/이수영_#27

twotwobread committed:
[FEAT] : 인증, 인가에 대한 custom 핸들러 추가.

- 인가에 대한 핸들러인 OAuthAccessDeniedHandler를 추가.
- 인증에 대한 핸들러인 OAuthAuthenticationEntryPoint 추가.
- 위 핸들러들을 SpringSecurityConfig에 설정 추가.
Close feature/이수영_#27

twotwobread committed:
[ADD] : jwt에서 이용되는 상수들을 추가.

- jwt에서 사용하는 value 값을 찾기 위한 key들을 추가.
CLose feature/이수영_#27

park-se-jun committed:
[FIX] : OAuthAuthenticationEntryPoint.java 삭제

- 커스텀을 잘못했는지 해당 필터를 설정해주면 제대로 동작하지 않아서 default를 사용하도록 OAuthAuthenticationEntryPoint.java 삭제

rhdtn311 committed:
[FIX] : 엔티티 전체 수정

rhdtn311 committed:
[FIX] : 엔티티 전체 수정 (#70)

## PR Description 🗒️
closed #63 
- DB 테이블 설계를 리팩토링 한대로 엔티티 클래스 전체 수정

## 추가 사항 및 설명 ➕

## 수정 사항 및 이유 🔄
- 프로젝트의 DB 테이블 설계를 변경했기 때문에

## To Reviewers 🙏

park-se-jun committed:
[FIX] : 변경된 엔티티와의 conflict 해소

- 변경된 엔티티와의 conflict 해소
- UserPrincipal 에서 UserDetails 제거
- UserPrincipal이 Users를 들고있는 것 변경 (Authority, Auttributes, Name 따로 들도록 변경)
- CustomOauth2UserService 에서 UserPrincipal로 넣어준 userRole을 UserService.join에서 사용하도록 코드 변경
- application.yml 일단 설정

park-se-jun committed:
[refactor] 62 로의 merge

twotwobread committed:
[RENAME] : CookieUtils에서 CokkieUtil로 파일이름 변경.

Close refactor/#71

twotwobread committed:
[DEL] : CustomUserPrincipal를 OAuth2User 구현만 수행하기 위해 UserDetails와 관련된 코드 삭제.

Close refactor/#71

twotwobread committed:
[CHORE] : 개행 추가

Close refactor/#71

twotwobread committed:
[ADD] : refresh token의 갱신을 위한 상수 추가.

Close refactor/#71

twotwobread committed:
[FEAT] : refresh 토큰을 갱신하기 위한 로직 추가.

- UserRepository에 user id와 refresh 토큰을 이용하여 User를 찾는 쿼리 메소드 추가.
- UserService에 들어온 access token, refresh token이 만료되었다면 이를 새로운 refresh token으로 갱신하는 메소드 추가.
- OAuthController에 token을 갱신하기 위한 handler 추가.
- UserService 필요없는 분기 삭제 및 예외 메시지 변경.
Close refactor/#71

twotwobread committed:
[REFACTOR] : 헤더로 부터 access token을 가져오는 메소드 Util 클래스로 분리.

Close refactor/#71

twotwobread committed:
[ADD] : token 반환을 위한 응답 dto 추가.

Close refactor/#71

park-se-jun committed:
[REFACTOR] : user 엔티티 변경

- notblank 가 enum에 걸려있으면 ddl문이 실행되지 않기에 notnull로 변경
- refresh token의 nullale 을 true 로 변경
- UserRole enum의 이름에 ROLE_ 을 붙임

twotwobread committed:
[FEAT] : 회원 상세정보 조회 (#86)

<!-- 이슈와 PR 이름을 맞춰주세요!! -->
<!-- 여러 라인에 걸쳐 문장이 필요한 경우, 각 라인마다 "-" 를 이용해주세요 -->

## PR Description 🗒️
<!-- 
PR에 대한 간략한 설명을 적어주세요
- #71 브랜치와 conflict 해결
## 추가 사항 및 설명 ➕
<!-- 
[FEAT] / [ADD]
추가 사항과 이에 대한 설명을 적어주세요
ex)
- 예매 기능 추가 
-->
- isDeleted를 고려하여 user id를 통해 유저를 조회할 수 있는 쿼리 메소드를 UserRepository에 추가.
- 유저 상세정보를 조회를 위한 UserService를 추가.
- 유저 상세정보를 조회를 위한 UserController를 추가.
close #12

Co-authored-by: So Hyeon <thgus345@naver.com>
Co-authored-by: rhdtn311 <68289543+rhdtn311@users.noreply.github.com>
Co-authored-by: 권성준 <tjdwns1865@naver.com>
Co-authored-by: Kong <rhdtn311@naver.com>

twotwobread committed:
[FEAT] : 회원 리스트 조회 (#87)

<!-- 이슈와 PR 이름을 맞춰주세요!! -->
<!-- 여러 라인에 걸쳐 문장이 필요한 경우, 각 라인마다 "-" 를 이용해주세요 -->

## PR Description 🗒️
<!-- 
PR에 대한 간략한 설명을 적어주세요
ex)
- 로그인 기능 추가.
-->
- 회원 리스트를 페이징하여 조회하는 기능 추가.
## 추가 사항 및 설명 ➕
<!-- 
[FEAT] / [ADD]
추가 사항과 이에 대한 설명을 적어주세요
ex)
- 예매 기능 추가
    - 
-->
- UserController의 회원 리스트 조회를 위한 핸들러 추가와 발생할 수 있는 예외에 대한 핸들러 추가.
- UserService의 회원 리스트 조회를 위한 메소드 추가.
- UserRepository에 isDeleted 상태를 고려한 회원 리스트 조회를 위해 쿼리 메소드 추가.

close #13

Co-authored-by: So Hyeon <thgus345@naver.com>
Co-authored-by: rhdtn311 <68289543+rhdtn311@users.noreply.github.com>
Co-authored-by: 권성준 <tjdwns1865@naver.com>
Co-authored-by: Kong <rhdtn311@naver.com>

twotwobread committed:
[FEAT] : log 관련 설정하기 (#98)

<!-- 이슈와 PR 이름을 맞춰주세요!! -->
<!-- 여러 라인에 걸쳐 문장이 필요한 경우, 각 라인마다 "-" 를 이용해주세요 -->

## PR Description 🗒️
<!-- 
PR에 대한 간략한 설명을 적어주세요
ex)
- 로그인 기능 추가.
-->
- 로그 관련한 의존성 추가, ignore 추가, logback 파일 추가.
## 추가 사항 및 설명 ➕
<!-- 
[FEAT] / [ADD]
추가 사항과 이에 대한 설명을 적어주세요
ex)
- 예매 기능 추가
    - 
-->
- 로그를 클라우드 와치로 쌓기 위하여 의존성 추가.
- logback-spring.xml 파일을 추가하여 출력될 로그문의 형태를 정하고 클라우드 와치로 로그를 보냄.
- gitignore을 설정하여 aws key 노출을 막음.
- [해당 내용 Notion
문서화](https://www.notion.so/backend-devcourse/CloudWatch-de832fc0baf04b0e8847e3005039efae)

close #95

Co-authored-by: So Hyeon <thgus345@naver.com>
Co-authored-by: rhdtn311 <68289543+rhdtn311@users.noreply.github.com>
Co-authored-by: 권성준 <tjdwns1865@naver.com>
Co-authored-by: Kong <rhdtn311@naver.com>

twotwobread committed:
[ADD] : gradle 의존성 추가.

twotwobread committed:
[ADD] : SpringSecurityConfig 파일 permitAll, User가 접근할 수 있는 url prefix 수정.

twotwobread committed:
[IMPROVE] : 회원 페이징 조회 로직 수정 (#117)

<!-- 이슈와 PR 이름을 맞춰주세요!! -->
<!-- 여러 라인에 걸쳐 문장이 필요한 경우, 각 라인마다 "-" 를 이용해주세요 -->
## PR Description 🗒️
close #107
<!-- 
PR에 대한 간략한 설명을 적어주세요
ex)
- 로그인 기능 추가.
-->
- 회원 페이징 조회 로직 수정 (먼저 count를 이용하여 가능한 PageRequest 생성)
## 추가 사항 및 설명 ➕
<!-- 
[FEAT] / [ADD]
추가 사항과 이에 대한 설명을 적어주세요
ex)
- 예매 기능 추가
    - 
-->
- UserRepository에 isDeleted가 false인 유저들을 count하는 쿼리 메소드 추가.
- UserService에서 유저 count하는 메소드 추가.
- UserController에서 유저를 count한 값을 PageService의 메소드를 이용하여 알맞은 PageRequest를
도출하고 이를 이용하여 유저 페이징 조회.

## To Reviewers 🙏
<!-- 
리뷰어들이 중점적으로 봐줬으면 하는 부분이나 
리뷰어들에게 추가적으로 하고 싶은 말을 적어주세요!! 
-->
- UserService에 countUser라는 메소드를 추가해서 컨트롤러에서 서비스의 메소드를 이용했는데 그냥
Repository에 있는 메소드를 쓰는 것이 나을까요??
- Controller에 있는 페이징 조회 메소드에도 Transactional이 있어야 한다고 생각하는데
Transactional이 필요할까요??
- 만약 필요하다면 하나의 layer를 따로 빼서 거기서 Transactional을 붙이는 것이 좋을까요 아니면 컨트롤러에 붙이는
것이 맞을까요? (controller에 Transactional을 붙이는 것이 익숙하지 않아 이상하다고 느껴서)

twotwobread committed:
[TEST] : 회원 조회에 대한 테스트 코드 추가 (#122)

<!-- 이슈와 PR 이름을 맞춰주세요!! -->
<!-- 여러 라인에 걸쳐 문장이 필요한 경우, 각 라인마다 "-" 를 이용해주세요 -->

## PR Description 🗒️
<!-- 
PR에 대한 간략한 설명을 적어주세요
ex)
- 로그인 기능 추가.
-->
close #88 
- 회원 조회 관련 테스트 코드 추가.
## 추가 사항 및 설명 ➕
<!-- 
[FEAT] / [ADD]
추가 사항과 이에 대한 설명을 적어주세요
ex)
- 예매 기능 추가
    - 
-->
- 테스트 컨벤션에 맞춰 테스트 코드 수정.
- UserRepository 테스트 코드 추가.
- UserService 테스트 코드 추가.
- UserController 테스트 코드 추가.

Co-authored-by: So Hyeon <thgus345@naver.com>
Co-authored-by: rhdtn311 <68289543+rhdtn311@users.noreply.github.com>
Co-authored-by: 권성준 <tjdwns1865@naver.com>
Co-authored-by: Kong <rhdtn311@naver.com>

twotwobread committed:
Merge branch 'develop' into refactor/#71

# Conflicts:
#	build.gradle
#	src/main/java/com/prgrms/be/intermark/common/exception/GlobalControllerAdvice.java
#	src/main/java/com/prgrms/be/intermark/common/service/page/PageService.java
#	src/main/java/com/prgrms/be/intermark/domain/casting/repository/CastingRepository.java
#	src/main/java/com/prgrms/be/intermark/domain/musical/model/Musical.java
#	src/main/java/com/prgrms/be/intermark/domain/musical/repository/MusicalRepository.java
#	src/main/java/com/prgrms/be/intermark/domain/musical_seat/model/MusicalSeat.java
#	src/main/java/com/prgrms/be/intermark/domain/schedule/controller/ScheduleController.java
#	src/main/java/com/prgrms/be/intermark/domain/schedule/model/Schedule.java
#	src/main/java/com/prgrms/be/intermark/domain/schedule/repository/ScheduleRepository.java
#	src/main/java/com/prgrms/be/intermark/domain/schedule/service/ScheduleService.java
#	src/main/java/com/prgrms/be/intermark/domain/schedule_seat/dto/ScheduleSeatResponseDTO.java
#	src/main/java/com/prgrms/be/intermark/domain/schedule_seat/dto/ScheduleSeatResponseDTOs.java
#	src/main/java/com/prgrms/be/intermark/domain/schedule_seat/model/ScheduleSeat.java
#	src/main/java/com/prgrms/be/intermark/domain/schedule_seat/repository/ScheduleSeatRepository.java
#	src/main/java/com/prgrms/be/intermark/domain/seat/repository/SeatRepository.java
#	src/main/java/com/prgrms/be/intermark/domain/ticket/controller/TicketController.java
#	src/main/java/com/prgrms/be/intermark/domain/ticket/dto/TicketCreateRequestDTO.java
#	src/main/java/com/prgrms/be/intermark/domain/ticket/model/Ticket.java
#	src/main/java/com/prgrms/be/intermark/domain/ticket/repository/TicketRepository.java
#	src/main/java/com/prgrms/be/intermark/domain/ticket/service/TicketService.java
#	src/main/java/com/prgrms/be/intermark/domain/user/User.java
#	src/main/java/com/prgrms/be/intermark/domain/user/repository/UserRepository.java
#	src/main/java/com/prgrms/be/intermark/domain/user/service/UserService.java
#	src/test/java/com/prgrms/be/intermark/common/page/PageResponseDTOTest.java

rhdtn311 reviewed (approved):
고생하셨습니다!!

Seongjun-Kwon commented on src/main/java/com/prgrms/be/intermark/auth/OAuth2AuthenticationSuccessHandler.java:
```diff
@@ -0,0 +1,65 @@
+    }
+
+    private String generateLoginSuccessJson(String accessToken) {
+        return "{\"token\":\"" + accessToken + "\"}";
```
다른 방식이 없을까요?

twotwobread replied:
ObjectMapper를 이용하여 json body 정보를 만들어서 response에 넣어주도록 수정하였습니다!!

Seongjun-Kwon commented on src/main/java/com/prgrms/be/intermark/domain/ticket/dto/TicketDeleteResponseDto.java:
```diff
@@ -0,0 +1,9 @@
+package com.prgrms.be.intermark.domain.ticket.dto;
+
+import com.prgrms.be.intermark.domain.ticket.TicketStatus;
+
+import lombok.Builder;
+
+@Builder
+public record TicketDeleteResponseDto(Long ticketId, TicketStatus status, String message) {
+}
```
이거는 현재 develop 에서는 없는 파일이에용

Seongjun-Kwon commented on src/main/resources/application.yml:
```diff
@@ -0,0 +1,23 @@
+  datasource:
+    driver-class-name: com.mysql.cj.jdbc.Driver
+    url: jdbc:mysql://localhost:3306/intermark
+    username: root
+    password: intermark123!
```
가리면 좋을 것 같아요

Seongjun-Kwon reviewed (approved):
고생 많으셨습니다~
현재 develop 이랑 버전이 다른 부분들이 좀 보입니다
merge 하기 전에 pull 해서 해결 부탁드려용

twotwobread committed:
Merge branch 'develop' into refactor/#71

twotwobread committed:
[DELETE] : 버전에 맞지 않는 필요없는 파일 삭제.

Close refactor/#71

twotwobread committed:
[CHORE] : 필요없는 설정파일 삭제 및 gitignore 수정.

twotwobread committed:
[DELETE] : 필요없는 설정파일 삭제.

twotwobread committed:
[DELETE] : yml 파일 삭제.

twotwobread committed:
[REFACTOR] : 문자열로 json body를 직접 만들어서 반환하던 부분을 object mapper 이용하도록 수정.

- 문자열로 직접 reponse에 json body를 넣어주던 부분을 Dto와 ObjectMapper를 이용하여 json body를 만들어주도록 변경.
Close refactor/#71

twotwobread committed:
Merge branch 'refactor/#71' of https://github.com/prgrms-be-devcourse/BE-03-InterMark into refactor/#71

park-se-jun commented on src/main/java/com/prgrms/be/intermark/common/dto/page/PageListIndexSize.java:
```diff
@@ -1,4 +1,4 @@
-package com.prgrms.be.intermark.common.dto.page.dto;
+package com.prgrms.be.intermark.common.dto.page;
```
PageListIndexSize와 PaeResponseDto는 develop 브랜치에서 common.dto.page.dto 안에 있으므로 그렇게 변경하는게 좋아 보입니다.(다른곳에서도 다 그렇게 참조해서 오류남)

park-se-jun commented on src/main/java/com/prgrms/be/intermark/common/dto/page/PageResponseDTO.java:
```diff
@@ -1,13 +1,12 @@
-package com.prgrms.be.intermark.common.dto.page.dto;
+package com.prgrms.be.intermark.common.dto.page;
```
PageListIndexSize와 PaeResponseDto는 develop 브랜치에서 common.dto.page.dto 안에 있으므로 그렇게 변경하는게 좋아 보입니다.(다른곳에서도 다 그렇게 참조해서 오류남)

park-se-jun commented on src/main/java/com/prgrms/be/intermark/domain/user/service/UserService.java:
```diff
@@ -1,26 +1,111 @@
+        Page<User> allUserPage = userRepository.findByIsDeletedFalse(pageable);
+        return new PageResponseDTO<>(allUserPage,
+                UserInfoResponseDTO::from,
+                PageListIndexSize.ADMIN_PERFORMANCE_LIST_INDEX_SIZE);
```
현재 코드에는 이곳에 해당하는 내용이 PageListIndexSize에 없습니다. 추가해 주세요. 짱수영님

park-se-jun commented on src/main/java/com/prgrms/be/intermark/domain/user/repository/UserRepository.java:
```diff
@@ -1,11 +1,24 @@
+
+    Optional<User> findByIdAndRefreshToken(Long userId, String refreshToken);
+
+    Optional<User> findByIdAndIsDeletedFalse(Long userId);
```
ticketService의 43번째 줄에 이 함수와 같은 역할을 하지만 메소드 명이 다른 함수가 있습니다. 2중 1개로 통일시켜 주세요.

https://github.com/prgrms-be-devcourse/BE-03-InterMark/blob/52ef3253151b59a31fa76ec0df0df286d5661fe9/src/main/java/com/prgrms/be/intermark/domain/ticket/service/TicketService.java#L43

park-se-jun commented on src/main/java/com/prgrms/be/intermark/domain/user/service/UserService.java:
```diff
@@ -1,26 +1,111 @@
+    }
+
+    @Transactional(readOnly = true)
+    public UserInfoResponseDTO findById(Long userId) {
```
이게 아까 성준이형한테 말한 부분! 
https://github.com/prgrms-be-devcourse/BE-03-InterMark/blob/52ef3253151b59a31fa76ec0df0df286d5661fe9/src/main/java/com/prgrms/be/intermark/domain/musical/service/MusicalFacadeService.java#L59
랑 일치하지 않는 return 값을 가지고 있네용.

아까 말 나온대로 facade 패턴을 위한 메소드를 하나 만들어서 그쪽 메소드의 이름을 그걸로다가 바꾸면 되지 않을까 싶습니당.

park-se-jun reviewed (changes requested):
일단 눈에보이는 오류들에 대해 적어봤습니다. 혼자 하기 힘들면 패프로 같이 ㄱ?

twotwobread committed:
[CHORE] : page dto 관련 import 수정.

twotwobread committed:
[ADD] : PageListIndexSize 관련 UserListIndexSize 추가.

Close refactor/#71

twotwobread committed:
[ADD] : Fasade 계층에서 사용하기 위한 findById 메소드를 UserService에 추가.

Close refactor/#71

twotwobread committed:
[DELETE] : GlobalControllerAdvice에서 핸들링하고 있어 EntityNotFoundException 관련 ExceptionHandler 삭제.

Close refactor/#71

park-se-jun commented on src/main/java/com/prgrms/be/intermark/common/exception/GlobalControllerAdvice.java:
```diff
@@ -6,71 +6,84 @@
 
+	@ResponseStatus(HttpStatus.NOT_FOUND)
+	@ExceptionHandler(EntityNotFoundException.class)
+	public ErrorResponse entityNotFoundExceptionHandler(EntityNotFoundException exception) {
```
29번째 줄과88번째 줄 2곳에서 쓰여서 컴파일시 에러가 발생합니다.

https://github.com/prgrms-be-devcourse/BE-03-InterMark/blob/e31af808cb71a3ef41d917ec166b40a9cc165c0c/src/main/java/com/prgrms/be/intermark/common/exception/GlobalControllerAdvice.java#L88

park-se-jun reviewed (commented):
핸들러 겹치는 것만 해결 하면 일단 문제는 없을 것 같기두 합니다. 제가 올렸던 application.yml을 좀 수정하긴 할듯

해당 변경사항과 별개로 uploadImageServiceImpl의 ${file.images} 여러가지 에로사항이 있는것 같던데 이건 그냥 application.yml에 저걸 바꾸기 전까지 file.images를 추가하는게 나을 것 같음

p.s gitignore에 application*.yml 있는디 기본적인 yml은 있는게 편할 거같아서 제꺼 pr에서 기본적인 yml 을 올리니 yml을 없는 현 상태에서 해당 구문을 빼는건 어떨지 건의 드립니다

twotwobread committed:
[DELETE] : 중복되는 EntityNotFoundException 관련 Handler를 삭제.

Close refactor/#71

twotwobread committed:
[CHORE] : gitignore 수정.

twotwobread commented:
> 핸들러 겹치는 것만 해결 하면 일단 문제는 없을 것 같기두 합니다. 제가 올렸던 application.yml을 좀 수정하긴 할듯
> 
> 해당 변경사항과 별개로 uploadImageServiceImpl의 ${file.images} 여러가지 에로사항이 있는것 같던데 이건 그냥 application.yml에 저걸 바꾸기 전까지 file.images를 추가하는게 나을 것 같음
> 
> p.s gitignore에 application*.yml 있는디 기본적인 yml은 있는게 편할 거같아서 제꺼 pr에서 기본적인 yml 을 올리니 yml을 없는 현 상태에서 해당 구문을 빼는건 어떨지 건의 드립니다

gitignore에서 뺐습니다!!

twotwobread committed:
Merge branch 'develop' into refactor/#71

# Conflicts:
#	src/main/java/com/prgrms/be/intermark/domain/user/User.java
